ocaml/typing/typeopt.ml

```diff
@@ -737,6 +737,8 @@ let rec layout_union l1 l2 =
   match l1, l2 with
   | Pbottom, l
   | l, Pbottom -> l
+  | Pnullable_value value_kind1, Pnullable_value value_kind2 ->
+      Pnullable_value (value_kind_union value_kind1 value_kind2)
```

Comment:
Shouldn't we also want a join between `Pnullable_value` and `Pvalue` to return something better than `Ptop`? This is necessary if the layouts are not incompatible.